File: src/main/java/fm/comunas/fm/building/model/enums/CodeBuildingElementDescription.java

package fm.comunas.fm.building.model.enums;

import fm.comunas.ddd.enums.model.base.EnumeratedBase;

public class CodeBuildingElementDescription extends EnumeratedBase {

	private final String category;

	public CodeBuildingElementDescription(CodeBuildingElementDescriptionEnum enumeration, String id, String name,
			String category) {
		super(enumeration, id, name);
		this.category = category;
	}

	public String getCategory() {
		return this.category;
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/building/model/enums/CodeBuildingElementDescription.kt
package io.zeitwert.fm.building.model.enums

import io.dddrive.enums.model.EnumeratedEnum
import io.dddrive.enums.model.base.EnumerationBase

enum class CodeBuildingElementDescription(
	override val defaultName: String,
	val category: String,
) : EnumeratedEnum {

	;

	override val enumeration get() = Enumeration

	companion object Enumeration :
		EnumerationBase<CodeBuildingElementDescription>(CodeBuildingElementDescription::class.java) {

		init {
			entries.forEach { addItem(it) }
		}

		fun getElementDescription(itemId: String?) = if (itemId != null) getItem(itemId) else null

	}
}
